tests/src/test/scala/akka/kafka/scaladsl/RebalanceExtSpec.scala

```diff
@@ -0,0 +1,373 @@
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.kafka.ConsumerMessage.{CommittableMessage, CommittableOffset}
+import akka.kafka._
+import akka.kafka.testkit.scaladsl.TestcontainersKafkaLike
+import akka.stream._
+import akka.stream.scaladsl.{Flow, Keep, Sink, Source}
+import akka.stream.testkit.scaladsl.StreamTestKit.assertAllStagesStopped
+import akka.{Done, NotUsed}
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.common.TopicPartition
+import org.scalatest._
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future, Promise}
+import scala.util.Try
+
+// https://github.com/akka/alpakka-kafka/pull/1263
+class RebalanceExtSpec extends SpecBase with TestcontainersKafkaLike with Inside {
+
+  implicit override val patienceConfig: PatienceConfig = PatienceConfig(30.seconds, 500.millis)
+
+  final val consumerClientId1 = "consumer-1"
+  final val consumerClientId2 = "consumer-2"
+
+  case class TopicPartitionMetaData(topics: List[String],
+                                    tps: List[TopicPartition],
+                                    tpFutureMap: Map[String, Promise[Done]],
+                                    producerTpsAck: Seq[Future[Done]],
+                                    messageStoreAndAck: Map[Int, MessageAck])
+
+  case class MessageAck(partitionName: String,
+                        messageCounter: AtomicInteger,
+                        waitUntil: Promise[Done],
+                        ackWaitUntil: Promise[Done])
+
+  def assignmentHandler(clientId: String): PartitionAssignmentHandler = new PartitionAssignmentHandler {
+    override def onAssign(tps: Set[TopicPartition], consumer: RestrictedConsumer): Unit = {
+      log.debug(s"assignmentHandler::onAssign: client_id $clientId tps $tps consumer $consumer ")
+    }
+    override def onRevoke(tps: Set[TopicPartition], consumer: RestrictedConsumer): Unit = {
+      log.debug(s"assignmentHandler::onRevoke: client_id $clientId tps $tps consumer $consumer ")
+    }
+    override def onLost(tps: Set[TopicPartition], consumer: RestrictedConsumer): Unit = {
+      log.debug(s"assignmentHandler::onLost: client_id $clientId tps $tps consumer $consumer ")
+    }
+    override def onStop(tps: Set[TopicPartition], consumer: RestrictedConsumer): Unit = {
+      log.debug(s"assignmentHandler::onStop: client_id $clientId tps $tps consumer $consumer ")
+    }
+  }
+
+  def consumerSettings(group: String,
+                       maxPollRecords: String,
+                       partitionAssignmentStrategy: String): ConsumerSettings[String, String] =
+    consumerDefaults
+      .withGroupId(group)
+      .withCloseTimeout(5.seconds)
+      .withPollInterval(300.millis)
+      .withPollTimeout(200.millis)
+      .withProperty(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, maxPollRecords) // 500 is the default value
+      .withProperty(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, partitionAssignmentStrategy)
```

Comment:
I see why `MAX_POLL_RECORDS_CONFIG` is important to this test, but close timeout, poll interval, and poll timeout seem rather arbitrary. Do you think these are necessary to reproduce the problem?

Reply:
@seglo I have removed those configuration which are not relevant. I have also removed few assertions for brevity. As I said earlier, the failure rate is not 100% (it is more like 85% on one machine vs. 50% on another). One can vary the poll interval to enhance the failure rate, for me it is about 400.ms. I'll take a look at the build failure, doesn't seem to be related with this change, I may need to re-merge the latest master.

Reply:
@seglo All except "Run tests with Scala 2.12 and AdoptOpenJDK 8" passed. Tests are also successful on my dev machine (for both scala_12 and scala_13 with 1.8 hotspot jvm). Can you please provide some guidance on how I should proceed. I also have doubts that the failures are related to my changes.

Reply:
The tests look fine. That was a flakey test that failed.